### PR TITLE
[tellstick] Adopt new API url

### LINKED ISSUE #15987: [tellstick] Binding doesn´t work any more.  Have to change api url to pa-api.telldus.com
Telldus live have change the api url to pa-api.telldus.com. 

## Expected Behavior
Same as before but api have to change to pa-api.telldus.com.

## Current Behavior
same as before

## Steps to Reproduce (for Bugs)

1.
2.

## Your Environment

* Version used: (e.g., openHAB and add-on versions)
* Environment name and version (e.g. Chrome 76, Java 8, Node.js 12.9, ...):
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...):

### LINKED ISSUE #16139: [tellstick] support add-on suggestions via USB finder
It looks like this binding depends on a USB stick to connect to OH (??) so in theory the binding could be automatically suggested via the core USB add-on suggestion finder. If that is the case, it simply needs the respective discovery-methods to be added to the binding's addon.xml file.

Signed-off-by: Andrew Fiddian-Green <software@whitebear.ch>

### PR DESCRIPTION
**Pull request by andrewfg**
Resolves #15987

Note: not (yet) tested.

Signed-off-by: Andrew Fiddian-Green <software@whitebear.ch>

### CODE REVIEW

**Comment by andrewfg** [comment]
Ping @ulfwin @dabj123

File: bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveDeviceController.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
     public static final long DEFAULT_INTERVAL_BETWEEN_SEND = 250;
     private static final int REQUEST_TIMEOUT_MS = 15000;
     private AsyncHttpClient client;
-    static final String HTTP_API_TELLDUS_COM_XML = "http://api.telldus.com/xml/";
+    static final String HTTP_API_TELLDUS_COM_XML = "http://pa-api.telldus.com/xml/"; // http://api.telldus.com/xml/"
```
Comment (wborn) [inline bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveDeviceController.java:74]: It's the responsibility of the version control system to keep track of historic code not the code itself:

```suggestion
    static final String HTTP_API_TELLDUS_COM_XML = "http://pa-api.telldus.com/xml/";
```

**Review by wborn — COMMENTED** [review]
It looks good to me except for this comment:

**Comment by andrewfg** [comment]
The jar file for testing is here..

https://github.com/andrewfg/openhab-addons/releases/download/v1/org.openhab.binding.tellstick-4.2.0-SNAPSHOT.jar

**Review by wborn — APPROVED** [review]
Thanks! LGTM now 👍

**Comment by dabj123** [comment]
How about? 

- Do not make more than 10 requests to pa-api.telldus.com per minute. 
- Double-check that your scripts do not attempt to contact the Telldus API again without waiting for at least 10 seconds. Retry without delay will result in HTTP 429 error.

**Comment by wborn** [comment]
Maybe you can create a PR for that @dabj123?
It's not easy to implement and test if you do not use Tellstick.

**Comment by andrewfg** [comment]
> Do not make more than 10 requests to pa-api.telldus.com per minute.

The polling rate of the binding is 10 seconds (so 6 calls per minute) .. so I presume that is Ok.

> do not attempt to contact the Telldus API again without waiting for at least 10 seconds
> It's not easy to implement and test if you do not use Tellstick.

I agree with @wborn :)

**Review by lsiepel — APPROVED** [review]
LGTM

Can this be cherry picked to 4.1.x?

**Comment by lolodomo** [comment]
Was this change tested by someone and reported as fixing the issues?

**Comment by jlaur** [comment]
> Can this be cherry picked to 4.1.x?

I don't think so. openhab/openhab-core#3922 was introduced only in 4.2. It would have been better with separate PR's for the URL bugfix and the USB suggestion enhancement. @andrewfg - will you be able to split the PR on short notice in order to possible get the URL fix into 4.1.1?

**Comment by jlaur** [comment]
> Was this change tested by someone and reported as fixing the issues?

Ping @dabj123, @ulfwin

**Comment by ulfwin** [comment]
I actually bought another base station using the local API as a work around, rather than Telldus live. If @dabj123 still uses live, it's probably easier to test for him.

**Comment by jlaur** [comment]
I checked https://telldus.com/driftinformation/ which is in Swedish (which is readable for me as Danish 🙂):
> Om du använder våra API-tjänster, vänligen notera följande:
>
> - Använd INTE api.telldus.com. Din IP kommer automatiskt att blockeras. Byt till pa-api.telldus.com om du inte redan har gjort det.
> - Gör inte mer än 10 förfrågningar till pa-api.telldus.com per minut.
> - Kontrollera och dubbelkolla att dina skript inte gör nya försök att kontakta Telldus API utan att vänta minst 10 sekunder. Försök på nytt utan fördröjning kommer innebära HTTP 429 error.

The first bullet "Använd INTE api.telldus.com. Din IP kommer automatiskt att blockeras." means: "do NOT use api.telldus.com. Your IP will automatically be blocked." So we can safely merge as it's pretty clear the old endpoint cannot be used anymore.

**Comment by andrewfg** [comment]
> which is in Swedish (which is readable for me as Danish

@jlaur yes I could read it too .. via Google Translate ;)

**Comment by dabj123** [comment]
I still use Telldus live bridge, but it seams to not work for me with new jar  "org.openhab.binding.tellstick-4.2.0-SNAPSHOT.jar" 

 Request [http://pa-api.telldus.com/xml/devices/list?supportedMethods=19] failed: ExecutionException HTTP request returned status code 401
2024-01-07 15:06:53.774 [WARN ] [ternal.live.TelldusLiveBridgeHandler] - Failed to update
org.openhab.binding.tellstick.internal.live.TelldusLiveException: HTTP request returned status code 401
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:299) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.updateDevices(TelldusLiveBridgeHandler.java:155) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.refreshDeviceList(TelldusLiveBridgeHandler.java:124) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.util.concurrent.ExecutionException: HTTP request returned status code 401
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:350) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:290) ~[?:?]
	... 8 more

**Comment by andrewfg** [comment]
> it seams to not work for me with new jar

@dabj123 an HTTP 401 error means 'authentication failed' so check your login credentials please.

**Comment by dabj123** [comment]
I know that, but I have double check every thing and generate new keys, but still recieve 401 or 429.

**Comment by lsiepel** [comment]
> I know that, but I have double check every thing and generate new keys, but still recieve 401 or 429.

Could you open a new issue to investigate that?

**Comment by andrewfg** [comment]
> but still recieve 401 or 429

@dabj123 these are TWO different issues..

- HTTP 401 is an authentication error
- HTTP 429 is because you are sending a retry request faster than 10 seconds _"Kontrollera och dubbelkolla att dina skript inte gör nya försök att kontakta Telldus API utan att vänta minst 10 sekunder. Försök på nytt utan fördröjning kommer innebära HTTP 429 error."_

I am guessing that if the HTTP 401 error is fixed, then there will not be any retries, so the HTTP 429 error will probably go away.

**Comment by dabj123** [comment]
Request [http://pa-api.telldus.com/xml/devices/list?supportedMethods=19] failed: ExecutionException HTTP request returned status code 401

<error> OAuth Verification Failed: Can't verify request, missing oauth_consumer_key or oauth_token </error>

**Comment by andrewfg** [comment]
> OAuth Verification Failed: Can't verify request, missing oauth_consumer_key or oauth_token

@dabj123 **_you_** are the TellStick user (apparently the only one) so **_only you_** know how to renew your user key or token. I cannot help you since I do not know the binding or the application.

**Comment by dabj123** [comment]
I have renew new keys etc. Maybe TellStick have new logic in the new api to handle OAuth Verification...

**Comment by andrewfg** [comment]
^
https://api.telldus.com/documentation/live

**Pull request merged**